File: ObjectOrientation/Node/Node.js

// CLASS: Node
//
// Author: Xian Mardiros, 7862786
//
// REMARKS: just a standard node implementation
//
//-----------------------------------------
"use strict"

class Node{
	#_item
	#_next

	constructor( item, next ){
		if( item == null || (next != null && ! next instanceof Node) ){
			throw "incorrect Node constructor format"
		}
		this.#_item = item
		this.#_next = next
	}

	get item(){
		return this.#_item
	}

	get next(){
		return this.#_next
	}

	set next( node ){
		if( node instanceof Node ){
			this.#_next = item
		}else{
			throw "Error: Node.next can only be set to an instance of Node\n"
		}
	}

	/*
	toString()
	recursively create a string of the contents of this node and all down-list nodes
	*/
	toString(){
		var string = this.#_item.toString()
		if( this.#_next != null ){
			string += ", " + this.#_next.toString()
		}
		return string
	}
}

module.exports = Node

File: ObjectOrientation/CPP/Node.cpp
// CLASS: Node
//
// Author: Xian Mardiros, 7862786
//
// REMARKS: Node implementation for queues, toString added

#include "Node.h"
#include <iostream>

using namespace std;

Node::Node() :item(nullptr), next(nullptr)  {}
Node::Node(ListItem *i, Node *n) :item(i), next(n) {}
Node::Node(ListItem *i) :item(i), next(nullptr) {}
Node *Node::getNext() {return next;}
void Node::setNext(Node *n) {next=n;}
ListItem *Node::getItem() {return item;}
void Node::setItem(ListItem* li) { item = li; }

/*
 * toString
 * PURPOSE: Give info about this and downstream nodes
 * RETURNS: a string with the desired info
 */
std::string Node::toString(){
    std::string retString = "";
    /* produces warning, uncomment for debugging
    //cout << this << endl;
    if( this == nullptr ){
        retString += "null";
    }else{
        //cout << item << endl;
        retString += item->toString() + " ";
        if( next != nullptr ){
            retString += next->toString();
        }else{
            retString += "\n";
        }
    }
    */
    return retString;
}
